winpr/libwinpr/synch/mutex.c

```diff
@@ -54,13 +54,14 @@ static int MutexGetFd(HANDLE handle)
 	return -1;
 }
 
-BOOL MutexCloseHandle(HANDLE handle) {
-	WINPR_MUTEX *mutex = (WINPR_MUTEX *) handle;
+BOOL MutexCloseHandle(HANDLE handle)
+{
+	WINPR_MUTEX* mutex = (WINPR_MUTEX*) handle;
 
 	if (!MutexIsHandled(handle))
 		return FALSE;
 
-	if(pthread_mutex_destroy(&mutex->mutex))
+	if (!pthread_mutex_destroy(&mutex->mutex))
```

Comment:
no, actually this was changed to prevent a leak. "If successful, the pthread_mutex_destroy() and pthread_mutex_init() functions shall return zero; otherwise, an error number shall be returned to indicate the error." this means that on success the handle was leaked. We could fix it to free it under failure and success.